plugins/modules/ndo_tenant_custom_qos_policy.py

```diff
@@ -0,0 +1,474 @@
+    - O(dscp_mappings.dscp_from) and O(dscp_mappings.dscp_to) cannot both be C(unspecified).
+    type: list
+    elements: dict
+    suboptions:
+      dscp_from:
+        description:
+        - The starting encoding point of the DSCP range.
+        - Defaults to unspecified when unset during creation.
+        type: str
+        choices:
+          - af11
+          - af12
+          - af13
+          - af21
+          - af22
+          - af23
+          - af31
+          - af32
+          - af33
+          - af41
+          - af42
+          - af43
+          - cs0
+          - cs1
+          - cs2
+          - cs3
+          - cs4
+          - cs5
+          - cs6
+          - cs7
+          - expedited_forwarding
+          - unspecified
+          - voice_admit
+        aliases: [ from ]
+      dscp_to:
+        description:
+        - The ending encoding point of the DSCP range.
+        - Defaults to unspecified when unset during creation.
+        type: str
+        choices:
+          - af11
+          - af12
+          - af13
+          - af21
+          - af22
+          - af23
+          - af31
+          - af32
+          - af33
+          - af41
+          - af42
+          - af43
+          - cs0
+          - cs1
+          - cs2
+          - cs3
+          - cs4
+          - cs5
+          - cs6
+          - cs7
+          - expedited_forwarding
+          - unspecified
+          - voice_admit
+        aliases: [ to ]
+      dscp_target:
+        description:
+        - The DSCP target encoding point for egressing traffic.
+        - Defaults to unspecified when unset during creation.
+        type: str
+        choices:
+          - af11
+          - af12
+          - af13
+          - af21
+          - af22
+          - af23
+          - af31
+          - af32
+          - af33
+          - af41
+          - af42
+          - af43
+          - cs0
+          - cs1
+          - cs2
+          - cs3
+          - cs4
+          - cs5
+          - cs6
+          - cs7
+          - expedited_forwarding
+          - unspecified
+          - voice_admit
+        aliases: [ target ]
+      target_cos:
+        description:
+        - The target CoS traffic type for the egressing traffic.
+        - Defaults to unspecified when unset during creation.
+        type: str
+        choices: [ background, best_effort, excellent_effort, critical_applications, video, voice, internetwork_control, network_control, unspecified ]
+      qos_priority:
+        description:
+        - The QoS priority level to which the DSCP values will be mapped.
+        - Defaults to unspecified when unset during creation.
+        type: str
+        choices: [ level1, level2, level3, level4, level5, level6, unspecified ]
+        aliases: [ priority, prio ]
+  cos_mappings:
+    description:
+    - The Class of Service (CoS) mappings of the Custom QoS Policy.
+    - O(cos_mappings.dot1p_from) and O(cos_mappings.dot1p_to) cannot both be C(unspecified).
```

Comment:
- Both O(cos_mappings.dot1p_from) and O(cos_mappings.dot1p_to) cannot be C(unspecified).